File: Selenium/selenium-todo-component-tester/src/test/java/no/ahlstroem/pages/EditPageObject.java
package no.ahlstroem.pages;

import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;
import org.openqa.selenium.support.PageFactory;

public class EditPageObject {
    
    private WebDriver webDriver;

    @FindBy(id="description-field")
    private WebElement descriptionField;

    @FindBy(id="save-button")
    private WebElement saveButton;

    @FindBy(id="delete-button")
    private WebElement deleteButton;

    public EditPageObject(WebDriver webDriver) {
        this.webDriver = webDriver;
        PageFactory.initElements(webDriver, this);
    }

    public EditPageObject fillDescription(String description) {
        descriptionField.clear();
        descriptionField.sendKeys(description);
        return this;
    }

    public HomePageObject clickDeleteButton() {
        deleteButton.click();
        return new HomePageObject(webDriver);
    }

    public HomePageObject clickSaveButton() {
        saveButton.click();
        return new HomePageObject(webDriver);
    }

}

File: todo-app/src/test/pageobjects/EditPageObject.ts
import { fireEvent, screen } from "@testing-library/react";
import HomePageObject from "./HomePageObject";

class EditPageObject {

private readonly descriptionField: HTMLElement;
private readonly saveButton: HTMLElement;
private readonly deleteButton: HTMLElement;

  constructor() {
    this.descriptionField = screen.getByRole('textbox', {name: 'Description'});
    this.saveButton = screen.getByRole('button', {name: 'Save'});
    this.deleteButton = screen.getByRole('button', {name: 'Delete'});
    this.verifyExistenceOfElements();
  }

  public fillDescription(description: string) {
    fireEvent.change(this.descriptionField, {target: {value: description}});
    return this;
  }

  public clickSaveButton() {
    fireEvent.click(this.saveButton);
    return new HomePageObject();
  }

  public clickDeleteButton() {
    fireEvent.click(this.deleteButton);
    return new HomePageObject();
  }

  private verifyExistenceOfElements() {
    if (!this.descriptionField) {
      fail('The description field has not been initialized');
    }
    if (!this.deleteButton) {
      fail('The cancel button has not been initialized');
    }
    if (!this.saveButton) {
      fail('The save button has not been initialized');
    }
  }
}

export default EditPageObject;